specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,577 @@
+  /// shared 2D texture of the TextureBuffer (handle/resource).
+
+  /// `timestampInMicroseconds` is video capture time with microseconds units.
+  /// The value does not have to be exact captured time, but it should be
+  /// increasing order because renderer (composition) ignores incoming
+  /// video frame (texture) if its timestampInMicroseconds is equal or prior to
+  /// the current compositing video frame. It also will be exposed to the
+  /// JS with `VideoFrame::timestamp`.
+  /// (https://docs.w3cub.com/dom/videoframe/timestamp.html).
+  HRESULT PresentBuffer([in] ICoreWebView2StagingTexture* buffer,
+    [in] UINT64 timestampInMicroseconds);
+  /// Stop streaming of the current stream id.
+  /// The Javascript will receive `MediaStreamTrack::ended` event when the API
+  /// is called.
+  /// The Javascript can restart the stream with getTextureStream.
+  /// The API call will release any internal resources on both of WebView2 host
+  /// and the browser processes.
+  /// API calls of Present, CreateBuffer will fail after this
+  /// with an error of COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED.
+  /// The Stop API will be called implicitly when ICoreWebView2StagingTextureStream
+  /// object is destroyed.
+  HRESULT Stop();
+  /// Event handler for those that occur at the Renderer side, the example
+  /// are CreateBuffer, Present, or Stop.
+  HRESULT add_ErrorReceived(
+      [in] ICoreWebView2StagingTextureStreamErrorReceivedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for ErrorReceived event.
+  HRESULT remove_ErrorReceived([in] EventRegistrationToken token);
+  /// Updates d3d Device when it is updated by RenderAdapterLUIDChanged
+  /// event.
+  HRESULT SetD3DDevice([in] IUnknown* d3dDevice);
+  /// Event handler for receiving texture by Javascript.
+  /// `window.chrome.webview.registerTextureStream` call by Javascript will
+  /// request sending video frame to the host where it will filter requested
+  /// page's origin against allowed origins. If allowed, the Javascript will
+  /// send a video frame (web texture), through MediaStreamTrack insertable APIs,
+  /// MediaStreamTrackGenerator.
+  /// https://www.w3.org/TR/mediacapture-transform/.
+  /// WebTextureReceived event will be called only when it receives
+  /// a web texture. There is no start event for receiving web texture.
+  HRESULT add_WebTextureReceived(
+      [in] ICoreWebView2StagingTextureStreamWebTextureReceivedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for WebTextureReceived event.
+  HRESULT remove_WebTextureReceived([in] EventRegistrationToken token);
+  /// Event handler for stopping of the receiving texture stream.
+  /// It is expected that the host releases any holding handle/resource from
+  /// the WebTexture before an event handler returns.
+  HRESULT add_WebTextureStreamStopped(

```

Comment:
We need to update this documentation to say what JavaScript APIs or situations result in this event getting raised. Its only during the call to unregister.